R/getPlot.R

```diff
@@ -0,0 +1,108 @@
+#'     "mad", "cv", "stddev", "geometricmean", "name", "none".
+#'   - gateLabelFontSize (double) Font size for gate labels.
+#'   - height (double) Image height. Defaults to 228.
+#'   - percentileStart (double) For contour plots, the percentile of the first
+#'     contour.
+#'   - percentileStep (double) For contour plots, the percentile step between
+#'     each contour.
+#'   - populationId (character) Population ID. Defaults to ungated.
+
+#'   - postSubsampleN (double) Randomly subsample the file to contain this many
+#'     events after gating.
+#'   - postSubsampleP (double) Randomly subsample the file to contain this
+#'     percent of events (0 to 1) after gating.
+#'   - renderGates (logical) Render gates into the image.
+#'   - seed (double) Seed for random number generator used for subsampling. Use
+#'     for deterministic (reproducible) subsampling. If omitted, a pseudo-random
+#'     value is used.
+#'   - smoothing (double) For density and contour plots, adjusts the amount of
+#'     smoothing. Defaults to 0 (no smoothing). Set to 1 for typical smoothing.
+#'     Higher values (up to 10) increase smoothing.
+#'   - strokeThickness (double) The thickness of histogram and contour plot
+#'     lines. Defaults to 1.
+#'   - tickLabelsQ logica Display tick labels. Defaults to false.
+#'   - ticksQ logical Display ticks. Defaults to true.
+#'   - width Integer Image width. Defaults to 228.
+#'   - xAxisLabelQ logical Display x axis label. Overrides axisLabelsQ.
+#'   - xAxisQ logical Display x axis line. Overrides axesQ.
+#'   - xTickLabelsQ logical Display x tick labels. Overrides tickLabelsQ.
+#'   - xTicksQ logical Display x ticks. Overrides ticksQ.
+#'   - yAxisLabel logical Display y axis label. Overrides axisLabelsQ.
+#'   - yAxisQ logical Display y axis line. Overrides axesQ.
+#'   - yTickLabelsQ logical Display y tick labels. Overrides tickLabelsQ.
+#'   - yTicksQ logical Display y ticks. Overrides ticksQ.
+#'
+#' @export
+#' @examples
+#' \dontrun{
+#' TODO
+#' getPlot(experimentId, fcsFileId, plotType, xChannel, yChannel)
+#' }
+getPlot <- function(experimentId,
+                    fcsFileId,
+                    plotType,
+                    xChannel,
+                    yChannel,
+                    zChannel = "",
+                    params = list(),
+                    file = "") {
+  checkDefined(experimentId)
+  experimentId <- lookupByName("experiments", experimentId)
+  checkDefined(fcsFileId)
+  fcsFileId <- lookupByName(paste("experiments", experimentId, "fcsfiles", sep = "/"), fcsFileId, "filename")
+
+  args <- list(
+    fcsFileId = fcsFileId,
+    plotType = plotType,
+    xChannel = xChannel,
+    yChannel = yChannel,
+    zChannel = zChannel
+  )
+
+  params <- c(args, params)
+
+  bytes <- baseGet(paste("experiments", experimentId, "plot", sep = "/"), params)
+
+  displayPlot <- function(b) {
+    if (requireNamespace("magick", quietly = TRUE)) {
+      magick::image_read(b)
```

Comment:
I don't think there's a reason to use `magick` here. That's basically used as a converter here.

`grid.raster` as used below will create a fuzzy image.

I think you can replace both with these three lines:
```
plot.new()
image <- png::readPNG(bytes, native=TRUE)
rasterImage(image, 0, 0, 1, 1, interpolate=FALSE)
```

Reply:
I find this to be pretty fuzzy as well: 
![image](https://user-images.githubusercontent.com/35822787/164198421-7f6df18e-d65d-4692-bda2-b504445efe32.png)

Reply:
I'll remove `magick` if you still think it's unnecessary. I sort of think that if a use has it already installed, it's likely that they'd find it convenient to already have a `magick` object, and it looks a little nicer.

Reply:
New proposal 2: Change `raw=FALSE` to `display=TRUE`, then `return(invisible(bytes))` on all paths.

Reply:
On all paths, or should we keep the bytes visible `if (raw)`?